### ISSUE TITLE
How did you derive your sampling algo?

### ISSUE THREAD

**Issue by jzhang38** [XiangLi1999/Diffusion-LM#39]
Hi Lisa, 

Thanks for your wonderful work.

May I ask how did you derive the sampling algo mathmatically for x_0 prediction?  (I am looking for the sort of proof given in DDPM regarding the e-prediction)

**Comment by XiangLi1999** [comment]
This is actually quite similar to the DDPM sampling algorithm. Both e-prediction and x_0 prediction will be transformed back to derive p(x_{t-1} | x_t), and both derivation rely on x_{t−1} =\sqrt{\alpha} f_\theta(x_t,t)+ \sqrt{1-\alpha} * N(0,1), where f_\theta(x_t,t) is the predicted x_0.  

I think reading the last paragraph of section 4.2 could help.

**Comment by jzhang38** [comment]
My confusion is that you appear to rely on the forward process q(x_{t-1}| x_0) to sample, whereas DDPM samples by predicting the mean of backward process p(x_{t-1} | x_t) (which we learn through the closed form solution of q(x_{t-1} | x_t, x_0)). Is there any deduction I can find (perhaps in other papers that also use x_0 prediction) to prove that these two samplings are mathematically equivalent?

In other words, DDPM samples through q(x_{t-1} | x_t, x_0), but Diffusion-LM samples through q(x_{t-1} |  f_\theta(x_t,t)).

**Comment by XiangLi1999** [comment]
Maybe checkout the last equation on page 17 of the Diffusion-LM ArXiv paper.

**Comment by jzhang38** [comment]
<img width="285" alt="Screenshot 2022-10-25 at 3 58 33 PM" src="https://user-images.githubusercontent.com/42993249/197716950-7e9b8280-0213-4c9f-8f1a-2f1c52822b51.png">
Thanks for your prompt reply!
Yeah I understand the training loss is essentially the same. My question is regarding the sampling algorithm. I think if we follow DDPM to perform sampling, we are supposed to sample with the mean as defined above, with x_0 predicted by f_\theta(x_t,t)
